django-stubs/test/client.pyi

```diff
@@ -128,25 +131,136 @@ class Client(ClientMixin, _RequestFactory[_MonkeyPatchedWSGIResponse]):
     ) -> None: ...
     # Silence type warnings, since this class overrides arguments and return types in an unsafe manner.
     def request(self, **request: Any) -> _MonkeyPatchedWSGIResponse: ...
-    def get(  # type: ignore
+    @overload  # type: ignore
+    def get(
```

Comment:
I think we can remove this overload, it is the same as `bool`

Reply:
@sobolevn this will cause a false positive tho for the default value, e.g. `Client().get('foo').response_chain` will be inferred since the `follow: Literal[True]` case matches and I can't put it behind the broader `follow: bool` case. The `client_follow_flag` test I added should also fail when the first override is deleted.

Reply:
Hmm. Ordinarily you would remove the default value (`= ...`) from the argument in one of the overloads, like

```python
@overload
def get(self, path: str, data: Any = ..., follow: Literal[True], secure: bool = ..., **extra: Any) -> _MonkeyPatchedWSGIResponseRedirect: ...
#                                         ^^^^^^^^^^^^^^^^^^^^^
@overload
def get(self, path: str, data: Any = ..., follow: bool = ..., secure: bool = ..., **extra: Any) -> _MonkeyPatchedWSGIResponse: ...
```

But here it's not allowed because it follows another argument with default `data`.

I'm not aware of any tricks to make it work. May indeed have to use 3 overloads.